### ISSUE TITLE
Bad autocorrect by Style/ArgumentsForwarding

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#12619]
Style/ArgumentsForwarding removes optional arguments when autocorrecting while not removing their use.

## Expected behavior

Given the following code:

```ruby
def local_nodes(type, ignored = [], &blk)
  ignored |= [:class, :module]
  exp.each_node(type, ignored, &blk)
end
```

I expect RuboCop to not change this code, or replace `&blk` with `&`, or replace `&blk` with `...` (but see #12618).

## Actual behavior

RuboCop changes the code to:

```ruby
def local_nodes(type, ...)
  ignored |= [:class, :module]
  exp.each_node(type, ignored, ...)
end
```

This makes `ignored` always equal to `true` when passed to the `#each_node` method.

## Steps to reproduce the problem

1. Create a file `foo.rb` with the content given in the first code block above.
2.  Run `rubocop -a`

## RuboCop version

```
$ bundle exec rubocop -V
1.60.0 (using Parser 3.3.0.3, rubocop-ast 1.30.0, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-performance 1.20.2
  - rubocop-rspec 2.26.1
```

**Comment by znz** [comment]
I can reproduce a similar issue with `TargetRubyVersion: '3.1'`.
And I cannot reproduce with `TargetRubyVersion: '3.2'`.

```
% docker run --rm -it ruby:3.1 /bin/bash
root@71a1d3a91de0:/# gem i rubocop
Fetching ast-2.4.2.gem
Fetching unicode-display_width-2.5.0.gem
Fetching parallel-1.24.0.gem
Fetching parser-3.3.0.4.gem
Fetching rubocop-ast-1.30.0.gem
Fetching rainbow-3.1.1.gem
Fetching ruby-progressbar-1.13.0.gem
Fetching rubocop-1.60.0.gem
Fetching regexp_parser-2.9.0.gem
Fetching language_server-protocol-3.17.0.3.gem
Successfully installed unicode-display_width-2.5.0
Successfully installed ruby-progressbar-1.13.0
Successfully installed ast-2.4.2
Successfully installed parser-3.3.0.4
Successfully installed rubocop-ast-1.30.0
Successfully installed regexp_parser-2.9.0
Successfully installed rainbow-3.1.1
Successfully installed parallel-1.24.0
Successfully installed language_server-protocol-3.17.0.3
Successfully installed rubocop-1.60.0
10 gems installed
root@71a1d3a91de0:/# cat > .rubocop.yml
AllCops:
  TargetRubyVersion: '3.1'
  NewCops: enable
root@71a1d3a91de0:/# cat > a.rb
def foo(a=false, &block)
  bar(a, 1, &block)
end
root@71a1d3a91de0:/# rubocop --only Style/ArgumentsForwarding -a a.rb
Inspecting 1 file
C

Offenses:

.rb:1:9: C: [Corrected] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
def foo(a=false, &block)
        ^^^^^^^^^^^^^^^
.rb:2:10: C: [Corrected] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
  bar(a, 1, &block)
         ^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
root@71a1d3a91de0:/# cat a.rb
def foo(...)
  bar(a, ...)
end
```

**Comment by mvz** [comment]
Thanks @znz, I didn't think to check what happens when targeting 3.2.

**Comment by koic** [comment]
The issue is caused by the same reason as #12618 and is planned to be resolved in #12624.
